source/Octopus.Tentacle.Tests.Integration/Util/Builders/Decorators/Proxies/MethodLoggingProxyDecorator.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Reflection;
+using System.Threading.Tasks;
+using Serilog;
+
+namespace Octopus.Tentacle.Tests.Integration.Util.Builders.Decorators.Proxies
+{
+    public class MethodLoggingProxyDecorator : ServiceProxy
+    {
+        readonly ILogger logger = new SerilogLoggerBuilder().Build().ForContext<MethodLoggingProxyDecorator>();
+        string serviceTypeName;
+
+        void Configure(string serviceTypeName)
+        {
+            this.serviceTypeName = serviceTypeName;
+        }
+
+        public static TService Create<TService>(TService service) where TService : class
+        {
+            var proxiedService = DispatchProxyAsync.Create<TService, MethodLoggingProxyDecorator>();
+            var proxy = proxiedService as MethodLoggingProxyDecorator;
+            proxy!.SetTargetService(service);
+            proxy.Configure(typeof(TService).Name);
+
+            return proxiedService;
+        }
+
+        protected override void OnStartingInvocation(MethodInfo targetMethod)
+        {
+            logger.Information("{ServiceName:l}.{MethodName:l}() started", serviceTypeName, targetMethod.Name);
+        }
+
+        protected override Task OnStartingInvocationAsync(MethodInfo targetMethod)
+        {
+            logger.Information("{ServiceName:l}.{MethodName:l}() started", serviceTypeName, targetMethod.Name);
```

Comment:
Can the async methods in this class call the sync method since the log entries are identical? If we don't want to call sync from async then a private method could be acceptable but may cause bloat?

Reply:
Yeah, I had thought about that... I had separated them as my thought was that when we remove the sync services from Tentacle, that the non-async `OnStartingInvocation`/`OnCompletingInvocation` would go away completely (as there will be no non-async calls to decorated services).